### PR TITLE
Adds new uncatchable flag for items

### PR DESCRIPTION
**Pull request by JamieD1**
Adds new uncatchable flag for items

### Intent of your Pull Request

Adds new uncatchable flag for items
Also smite pie is not not catchable, however mass pie is still catchable

#### Changelog

:cl:  
rscadd: Adds new uncatchable flag for items
/:cl:

### CODE REVIEW

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -67,6 +67,7 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 	var/strip_delay = 40 //In deciseconds, how long an item takes to remove from another person
 	var/breakouttime = 0
 	var/list/materials
+	var/uncatchable = 0
```
Comment (Amelia0010) [inline code/game/objects/items.dm]: :}

File: code/modules/food_and_drinks/food/snacks_pie.dm [hunk 2]
```diff
@@ -64,6 +64,7 @@
 
 /obj/item/reagent_containers/food/snacks/pie/cream/admin
 	pass_flags = PASSTABLE | PASSGLASS | PASSGRILLE | PASSBLOB | PASSCLOSEDTURF | LETPASSTHROW | PASSDOOR | PASSMACHINES | PASSCOMPUTER
+	uncatchable = 1
```
Comment (Amelia0010) [inline code/modules/food_and_drinks/food/snacks_pie.dm]: :)

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 3]
```diff
@@ -54,6 +54,8 @@
 		if(can_catch_item())
 			if(istype(AM, /obj/item))
 				var/obj/item/I = AM
+				if(I.uncatchable == 1)
```
Comment (Amelia0010) [inline code/modules/mob/living/carbon/carbon_defense.dm]: :)

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 4]
```diff
@@ -54,6 +54,8 @@
...
+					return 0
```
Comment (Amelia0010) [inline code/modules/mob/living/carbon/carbon_defense.dm]: :)

**Review by Amelia0010 — CHANGES_REQUESTED** [review]
what even is a boolean

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 5]
```diff
@@ -54,12 +54,14 @@
 		if(can_catch_item())
 			if(istype(AM, /obj/item))
 				var/obj/item/I = AM
+				if(I.uncatchable == FALSE)
```
Comment (SomeguyManperson) [inline code/modules/mob/living/carbon/carbon_defense.dm]: What

**Review by SomeguyManperson — DISMISSED** [review]
also undocumented changes

**Comment by patpol4** [comment]
fuck i misclicked and deleted the branch, thanks github mobile app

**Comment by JamieD1** [comment]
@SomeguyManperson Care to explain?

**Review by alexkar598 — DISMISSED** [review]
Use proper flags, dont create another variable

File: code/modules/mob/living/carbon/carbon_defense.dm (hunk 5 above)
Comment (JamieD1) [reply code/modules/mob/living/carbon/carbon_defense.dm]: ?

**Comment by JamieD1** [comment]
> also undocumented changes

?

**Comment by JamieD1** [comment]
> Use proper flags, dont create another variable

Reee I knew you were going to ask that

**Review by Amelia0010 — APPROVED** [review]

**Pull request merged**